main/main_loop.c

```diff
@@ -621,17 +621,24 @@ main_task_timer_sig_check_for_fw_updates_stop(void)
 }
 
 void
-main_task_start_timer_after_hotspot_activation(void)
+main_task_start_timer_hotspot_deactivation(void)
 {
-    LOG_INFO("### Start timer after Wi-Fi AP activation (%u seconds)", MAIN_TASK_TIMEOUT_AFTER_HOTSPOT_ACTIVATION_SEC);
-    os_timer_sig_one_shot_start(g_p_timer_sig_after_wifi_ap_activation);
+    LOG_INFO("### Start Wi-Fi AP deactivation timer (%u seconds)", MAIN_TASK_TIMEOUT_HOTSPOT_DEACTIVATION_SEC);
+    os_timer_sig_one_shot_start(g_p_timer_sig_deactivate_wifi_ap);
 }
 
 void
-main_task_stop_timer_after_hotspot_activation(void)
+main_task_stop_timer_hotspot_deactivation(void)
 {
-    LOG_INFO("### Stop timer after Wi-Fi AP activation");
-    os_timer_sig_one_shot_stop(g_p_timer_sig_after_wifi_ap_activation);
+    LOG_INFO("### Stop Wi-Fi AP deactivation timer");
+    os_timer_sig_one_shot_stop(g_p_timer_sig_deactivate_wifi_ap);
+}
+
+void
+main_task_send_sig_to_stop_wifi_hotspot(void)
+{
+    LOG_INFO("Force activate WiFi hotspot deactivation timer");
+    os_timer_sig_one_shot_simulate(g_p_timer_sig_deactivate_wifi_ap);
```

Comment:
Yes, we need to call os_timer_sig_one_shot_simulate here because it will trigger the timer immediately, but os_timer_sig_one_shot_start will trigger it only after a delay.